src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,223 +33,335 @@
-        region_data = _download_zip(
+        output_dir_country = self._download(
+            url=url_country_date,
+            area=self._iso2,
+            pub_year=pub_year,
+            pub_month=pub_month,
+        )
+        return output_dir_country
+
+    def _download_region(
+        self,
+        pub_year: int,
+        pub_month: str,
+        clobber: bool,
+    ) -> Optional[Path]:
+        """
+        Download fewsnet data that covers the region the iso3 belongs to.
+
+        Returns
+        -------
+        region_data : Path
+            If region data exists, return the saved dir else return None
+        """
+        url_region_date = _BASE_URL_REGION.format(
+            region_code=self._region_code,
+            region_name=self._region_name,
+            YYYY=pub_year,
+            MM=pub_month,
+        )
+
+        output_dir_region = self._download(
             url=url_region_date,
-            zip_filename=zip_filename_region,
-            output_dir=output_dir_region,
+            area=self._region_code,
+            pub_year=pub_year,
+            pub_month=pub_month,
+        )
+        return output_dir_region
+
+    def _download(self, url, area, pub_year, pub_month):
+        """
+        Define output names and call _download_zip.
+
+        url: str
+            URL the zip file is located
+        area: str
+            Identifier of which area the data covers. This is either the ISO2
+            or the region code
+        pub_year: int
+            publication year of the data that should be downloaded
+        pub_month: int
+            publication month of the data that should be downloaded. This
+            commonly refers to the month of the Current Situation period
+        """
+        # filenames have upper iso2/regioncode, so use that for dirs as well
+        output_dir = self._get_raw_dir_date(
+            area=area, pub_year=pub_year, pub_month=pub_month
+        )
+        _download_zip(
+            filepath=output_dir,
+            zip_filename=self._get_zip_filename(
+                area=area, pub_year=pub_year, pub_month=pub_month
+            ),
+            url=url,
         )
-    else:
-        region_data = output_dir_region
-    return region_data
+        return output_dir
 
+    def _get_raw_dir_date(self, area, pub_year, pub_month):
+        return self._raw_base_dir / f"{area}_{pub_year}{pub_month}"
 
+    def _get_zip_filename(self, area, pub_year, pub_month):
+        return f"{area}{pub_year}{pub_month}.zip"
+
+    def _find_raw_dir_date(self, pub_year, pub_month):
+        """
+        Check if a dir exists for the given `pub_year`-`pub_month`.
+
+        Should either cover the iso2 or region.
+        If exists, returns the dir path.
+        """
+        country_dir = self._get_raw_dir_date(
+            area=self._iso2, pub_year=pub_year, pub_month=pub_month
+        )
+        if country_dir.is_dir():
+            return country_dir
+        else:
```

Comment:
You can remove the else, that way the nesting is reduced a bit

Reply:
hahaha nesting here we are. Should come up with a meme about it

Reply:
But I find not nesting very hard, so can you double-check if this is what you meant?

Reply:
Aha yes, I think it's best to avoid having the main body of a function / method nested at all, if possible. And in this case your second return is 2x nested.